### PR TITLE
[MOB-11441] Add CodePush Support

### PR DESCRIPTION
**Pull request by DavidMina96**
## Description of the change

- Add `codePushLabel` property to `InstabugConfig` and its native mappings on both platforms, along with the needed tests.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Issue links go here

### Development

- [x] Lint rules pass locally
- [x] The code changed/added as part of this pull request has been covered with tests

### Code review

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: example/ios/InstabugSampleTests/InstabugSampleTests.m [hunk 1]
```diff
@@ -218,6 +219,15 @@ - (void)testSetReproStepsMode {
   OCMVerify([mock setReproStepsMode:reproStepsMode]);
 }
 
+- (void)testSetCodePushVersion {
+  id mock = OCMClassMock([Instabug class]);
+   NSString *codePushLabel = @"v123";
+
+   OCMStub([mock setCodePushVersion:codePushLabel]);
+   [self.instabugBridge setCodePushVersion:codePushLabel];
+   OCMVerify([mock setCodePushVersion:codePushLabel]);
```
Comment (a7medev) [inline example/ios/InstabugSampleTests/InstabugSampleTests.m:228]: Small formatting suggestion to keep the spacing consistent.

```suggestion
  NSString *codePushLabel = @"v123";

  OCMStub([mock setCodePushVersion:codePushLabel]);
  [self.instabugBridge setCodePushVersion:codePushLabel];
  OCMVerify([mock setCodePushVersion:codePushLabel]);
```

File: android/src/main/java/com/instabug/reactlibrary/RNInstabugReactnativeModule.java [hunk 2]
```diff
@@ -134,7 +134,7 @@ public void run() {
      * @param invocationEventValues The events that invoke the SDK's UI.
      */
     @ReactMethod
-    public void init(final String token, final ReadableArray invocationEventValues, final String logLevel) {
+    public void init(final String token, final ReadableArray invocationEventValues, final String codePushLabel, final String logLevel) {
```
Comment (a7medev) [inline android/src/main/java/com/instabug/reactlibrary/RNInstabugReactnativeModule.java:137]: I wonder where the word `label` came from as we already use `version` on both iOS and Android, is there a reason why it's not just `codePushVersion`?

File: example/ios/InstabugSampleTests/InstabugSampleTests.m [hunk 3]
```diff
@@ -66,12 +66,13 @@ - (void)testInit {
...
   IBGSDKDebugLogsLevel sdkDebugLogsLevel = IBGSDKDebugLogsLevelDebug;
   
   XCTestExpectation *expectation = [self expectationWithDescription:@"Testing [Instabug init]"];
   
   OCMStub([mock startWithToken:appToken invocationEvents:floatingButtonInvocationEvent]);
-  [self.instabugBridge init:appToken invocationEvents:invocationEvents debugLogsLevel:sdkDebugLogsLevel];
+  [self.instabugBridge init:appToken invocationEvents:invocationEvents codePushLabel:codePushLabel debugLogsLevel:sdkDebugLogsLevel];
 
   [[NSRunLoop mainRunLoop] performBlock:^{
     OCMVerify([mock startWithToken:appToken invocationEvents:floatingButtonInvocationEvent]);
```
Comment (a7medev) [inline example/ios/InstabugSampleTests/InstabugSampleTests.m:78]: I think we also need to verify that the `setCodePushVersion:` has been called with the correct `codePushLabel` in the `init` test to ensure that it does indeed call it.

File: tests/modules/Instabug.spec.js [hunk 4]
```diff
@@ -149,6 +150,7 @@ describe('Instabug Module', () => {
     expect(NativeInstabug.init).toBeCalledWith(
       instabugConfig.token,
       instabugConfig.invocationEvents,
+      instabugConfig.codePushLabel,
```
Comment (a7medev) [inline tests/modules/Instabug.spec.js:153]: I think we should also test the case when the `codePushLabel` is not passed and verify that it was called with an empty string `''`.

Comment (DavidMina96) [reply tests/modules/Instabug.spec.js:153]: Makes sense! I added another test for `init` to verify the native method is called with the default values given the optional properties aren't passed.

**Review by a7medev — APPROVED** [review]
LGTM 👏
